basics/section_2-40-42.rb

```diff
@@ -0,0 +1,45 @@
+# classes must always start with Capital in Ruby
+class Student
+  attr_accessor :first_name, :last_name, :email, :username, :password
+
+  # attr_reader # --> for reading only
+
+  # instance variable starts with @
+  # @first_name
+  # @last_name
+  # @email
+  # @username
+  # @password
+
+  # this method is run everytime a new instance is initialized
+  def initialize(firstname, lastname, username, email, password)
```

Comment:
The technical term for this is `constructor`.
Read up on `constructor` in class in OOP to get a better understanding